### PR TITLE
MWPW-136372: Removing the loading screen to improve performance scores

### PR DESCRIPTION
**Pull request by echampio-at-adobe**
* Removing the loading screen to improve performance scores. 

Resolves: [MWPW-136372](https://jira.corp.adobe.com/browse/MWPW-136372)

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.page/drafts/delta/app-recommender-test/?martech=off
- After: https://mwpw-136372--milo--echampio-at-adobe.hlx.page/drafts/xiasun/quiz-2/?martech=off

### CODE REVIEW

File: libs/blocks/quiz/quiz.css [hunk 1]
```diff
@@ -11,7 +11,7 @@
 }
 
 .quiz {
-  min-height: calc(100vh - 131px); /* window height minus global nav + global footer heights */
+  min-height: calc(100vh - 64px); /* window height minus global nav height */
```
Comment (mokimo) [inline libs/blocks/quiz/quiz.css:14]: Do we have a var for this?
Also the global navigation can be either with or without breadcrumbs, which changes the height

File: libs/blocks/quiz/quiz.js [hunk 2]
```diff
@@ -254,7 +254,7 @@ const App = ({
 
   useEffect(() => {
     const getStringValue = (propName) => {
-      if (!selectedQuestion) return '';
+      if (!selectedQuestion || !selectedQuestion.questions) return '';
```
Comment (mokimo) [inline libs/blocks/quiz/quiz.js:257]:
```suggestion
      if (!selectedQuestion?.questions) return '';
```

File: libs/blocks/quiz/quiz.js [hunk 3]
```diff
@@ -264,11 +264,8 @@ const App = ({
     }
   }, [selectedQuestion, stringQList]);
 
-  if (!isDataLoaded || !selectedQuestion) {
-    return html`<div class="quiz-load">Loading</div>`;
-  }
-
   const getStringValue = (propName) => {
+    if (!selectedQuestion || !selectedQuestion.questions) return '';
```
Comment (mokimo) [inline libs/blocks/quiz/quiz.js:268]:
```suggestion
      if (!selectedQuestion?.questions) return '';
```

File: libs/blocks/quiz/quiz.js [hunk 4]
```diff
@@ -279,43 +276,48 @@ const App = ({
...
-                      onOptionClick=${onOptionClick}
-                      getOptionsIcons=${getOptionsIcons}
-                      handleOnNextClick=${handleOnNextClick}
-                      btnAnalyticsData=${btnAnalytics}/>
+                    <${DecorateBlockForeground} 
+                        heading=${getStringValue('heading')} 
+                        subhead=${getStringValue('sub-head')} 
+                        btnText=${getStringValue('btn')} />
+                  
+                    ${selectedQuestion && selectedQuestion.questions && html`<${GetQuizOption}
```
Comment (mokimo) [inline libs/blocks/quiz/quiz.js:296]:
```suggestion
                    ${selectedQuestion?.questions && html`<${GetQuizOption} 
```

File: libs/blocks/quiz/quiz.js [hunk 5]
```diff
@@ -279,43 +276,48 @@ const App = ({
     return optionItem && optionItem[prop] ? optionItem[prop] : '';
   };
 
-  return html`<div class="quiz-container">
-                  <${StepIndicator} 
-                    currentStep=${currentStep} 
-                    totalSteps=${totalSteps} 
-                    prevStepIndicator=${prevStepIndicator}
-                    top="${true}" />
+  if (isDataLoaded) {
```
Comment (mokimo) [inline libs/blocks/quiz/quiz.js:279]: If you do any early return, you don't need to put this into an else/if
```suggestion
  if(!isDataLoaded) return ''
```

**Review by mokimo — APPROVED** [review]
Looks good after a couple of nits are addressed!

File: libs/blocks/quiz/quiz.css (hunk 1 above)
Comment (overmyheadandbody) [reply libs/blocks/quiz/quiz.css:14]: There are actually some variables that could be used, defined here - https://github.com/adobecom/milo/blob/main/libs/styles/styles.css#L4-L6

**Comment by mokimo** [comment]
FYI consider the blackout period before merging.

File: libs/blocks/quiz/quiz.css (hunk 1 above)
Comment (echampio-at-adobe) [reply libs/blocks/quiz/quiz.css]: Thanks for the pointer to styles.css. I imported it take advantage of the variable you mentioned. The quiz designs always request an absolutely minimal header, so I believe we're on with the assumption we're making about the height, especially not that we can take advantage of that shared variable.

File: libs/blocks/quiz/quiz.js (hunk 5 above)
Comment (echampio-at-adobe) [reply libs/blocks/quiz/quiz.js]: Done. Thanks!

**Review by fullcolorcoder — APPROVED** [review]

**Comment by echampio-at-adobe** [comment]
> FYI consider the freeze period before merging.

Hi @mokimo. We're just merging to the uar-integration branch for the near future, not to main, so we won't perturb the freeze for MAX.

File: libs/blocks/quiz/quiz.css [hunk 6]
```diff
@@ -11,7 +13,7 @@
 }
 
 .quiz {
-  min-height: calc(100vh - 131px); /* window height minus global nav + global footer heights */
+  min-height: calc(100vh - var(--global-height-nav)); /* window height minus global nav height from styles.css */
```
Comment (colloyd) [inline libs/blocks/quiz/quiz.css:16]: I'm seeing a permanent scrollbar with these values in chrome, however, if I add a -1px after the var, it seems to resolve it. Maybe we need to adjust this a bit. 

Also, for what it's worth, Chris Millar removed this min-height stuff in his quiz-perf branch due to some cls: 

https://github.com/adobecom/milo/blob/quiz-perf/libs/blocks/quiz/quiz.css#L12

**Review by colloyd — CHANGES_REQUESTED** [review]

**Pull request closed without merging**